src/main/riddl/Tenant/tenantTypeDefinitions.riddl

```diff
@@ -1,3 +1,10 @@
+type OptionalInfo is {
+  name: String?,
+  primaryContact: Contact?,
+  address: Address?,
+  orgs: Organization*
```

Comment:
Another way to model memvbers and organizations here is as a query on a projection, provided as a function, because each member and organization references its owning tenant so the projection should be able to get them for use in tenant by something like "SELECT id from members" and similarly for organizations.

Reply:
Orgs is already in Info of Tenant, this is just the version of Info with all fields Optional. I agree a projection may be more appropriate, but at the same time it may make sense to have Organizations in Tenant Info just in case there are scenarios when we want to get Orgs directly from Tenant without having to go to a View (I'm not sure why though...)

Reply:
I was thinking the required fields could easily be discerned, but not directly from the OptionalInfo, rather by comparing to Info. Since Organizations are `*` and not `+` in Info, it is not a required field. That may not be obvious enough though?

Reply:
Yeah, totally not obvious enough. But more importantly, it blurs the state transition. How many organizations are needed before the entity can transition from Draft to Established?  1? 10? 100?  Its best not to deal with these kinds of issues and recognize in Draft state that organizations are not essential.  If this entity wanted to auto-transition to established, it would have to have some minimum number of "orgs" needed.  With merely optional fields, there are only two cardinalities: 0 (not complete) and 1 (complete). So, when all optional fields are "complete", the entity can auto-transition to Established.

Reply:
So are you recommending that for `DraftState`s we have 2 Info objects, one for Required fields and the other for Optional? And then we combine when creating Info Established or other subsequent state?

Reply:
Right, I was only considering the lazy case here :)